src/PowerShellEditorServices.Hosting/HostLogger.cs

```diff
@@ -0,0 +1,333 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Serilog;
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Management.Automation.Host;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerShell.EditorServices.Hosting
+{
+    /// <summary>
+    /// User-facing log level for editor services configuration.
+    /// The underlying values of this enum align to both Microsoft.Logging.Extensions.LogLevel
+    /// and Serilog.Events.LogEventLevel.
+    /// </summary>
+    public enum PsesLogLevel
+    {
+        Diagnostic = 0,
+        Verbose = 1,
+        Normal = 2,
+        Warning = 3,
+        Error = 4,
+    }
+
+    /// <summary>
+    /// A logging front-end for host startup allowing handover to the backend
+    /// and decoupling from the host's particular logging sink.
+    /// </summary>
+    public class HostLogger :
+        IObservable<(PsesLogLevel logLevel, string message)>,
+        IObservable<(int logLevel, string message)>
+    {
+        /// <summary>
+        /// A simple translation struct to convert PsesLogLevel to an int for backend passthrough.
+        /// </summary>
+        private struct LogObserver : IObserver<(PsesLogLevel logLevel, string message)>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Implement 'IEquatable<T>' in value type 'LogObserver'.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)